packages/api/src/record.ts

```diff
@@ -304,14 +444,55 @@ export class Record implements RecordModel {
    *
    * @beta
    */
-  async send(target: string): Promise<ResponseStatus> {
-    const { reply: { status } } = await this._agent.sendDwnRequest({
-      messageType    : DwnInterfaceName.Records + DwnMethodName.Write,
-      author         : this._connectedDid,
-      dataStream     : await this.data.blob(),
-      target         : target,
-      messageOptions : this.toJSON(),
-    });
+  async send(_options?: string | { [key: string]: any }): Promise<ResponseStatus> {
+
+    const initialWrite = this._initialWrite;
+    const options = !_options ? { target: this._connectedDid } : typeof _options === 'string' ? { target: _options } : _options;
+
+    if (!options.target){
+      options.target = this._connectedDid;
+    }
+
+    if (initialWrite && !Record.checkSendCache(this._recordId, options.target)){
+
+      const initialState = {
+        messageType : DwnInterfaceName.Records + DwnMethodName.Write,
+        author      : this._connectedDid,
+        target      : options.target,
+        rawMessage  : removeUndefinedProperties({
+          contextId: this._contextId,
+          ...initialWrite
+        })
+      } as any;
+
+      await this._agent.sendDwnRequest(initialState);
+
+      Record.setSendCache(this._recordId, options.target);
+
+    }
+
+    const latestState = {
+      messageType : DwnInterfaceName.Records + DwnMethodName.Write,
+      author      : this._connectedDid,
+      dataStream  : await this.data.blob(),
+      target      : options.target
+    } as any;
+
+    if (this._authorization) {
+      latestState.rawMessage = removeUndefinedProperties({
+        contextId     : this._contextId,
+        recordId      : this._recordId,
+        descriptor    : this._descriptor,
+        attestation   : this._attestation,
+        authorization : this._authorization,
+        encryption    : this._encryption,
+      });
+    }
+    else {
+      latestState.messageOptions = this.toJSON();
+    }
```

Comment:
Please add inline comments that explain what is going on here so its easier to maintain in the future.

Reply:
Done